property.go

```diff
@@ -14,6 +14,30 @@ type Property interface {
 	GetType() PropertyType
 }
 
+type PropertyArray []Property
```

Comment:
Not sure if this is a good name or not. `Properties` was already taken by `map[string]Property` so this seemed like a good next choice.